sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/OffHeapColumnVector.java

```diff
@@ -84,6 +84,24 @@ public long valuesNativeAddress() {
     return data;
   }
 
+  @Override
+  public void reset() {
+    super.reset();
+    if (!isConstant && hugeVectorThreshold > 0 &&
+        capacity > hugeVectorThreshold) {
+      capacity = defaultCapacity;
+      Platform.freeMemory(nulls);
+      Platform.freeMemory(data);
+      Platform.freeMemory(lengthData);
+      Platform.freeMemory(offsetData);
+      nulls = 0;
+      data = 0;
+      lengthData = 0;
+      offsetData = 0;
```

Comment:
shall we add a new method `releaseMemory` to share the code between `reset` and `close`?